packages/api-graphql/src/GraphQLAPI.ts

```diff
@@ -283,6 +285,13 @@ export class GraphQLAPIClass {
 			graphql_endpoint_iam_region: customEndpointRegion,
 		} = this._options;
 
+		let userAgentDetails = customUserAgent;
+		if (!userAgentDetails) {
+			userAgentDetails = { category: Category.API };
+		} else {
+			userAgentDetails.category = userAgentDetails.category ?? Category.API;
+		}
```

Comment:
Do we mean to update the reference from the parameter directly? Generally speaking, this is not desirable so I want to check if it is intentional

If not intentional I'd suggest just spreading the param so we're not mutating the param, e.g.
```suggestion
		const userAgentDetails = { category: Category.API, ...customUserAgent };
```